File: rock63/src/main/java/com/uksusoff/rock63/ui/BaseActivity.java
package com.uksusoff.rock63.ui;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import androidx.annotation.StringRes;
import androidx.appcompat.app.AppCompatActivity;

import com.j256.ormlite.android.apptools.OpenHelperManager;
import com.uksusoff.rock63.data.DBHelper;
import com.uksusoff.rock63.data.InternalPrefs_;

import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.UiThread;
import org.androidannotations.annotations.sharedpreferences.Pref;

@EActivity
public abstract class BaseActivity extends AppCompatActivity {

    public static final String ACTION_CHECK_ALARM = "com.uksusoff.rock63.ui.ACTION_CHECK_ALARM";

    private DBHelper dbHelper = null;

    @Pref
    InternalPrefs_ internalPrefs;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Intent intent = new Intent(ACTION_CHECK_ALARM);
        sendBroadcast(intent);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (dbHelper != null) {
            OpenHelperManager.releaseHelper();
            dbHelper = null;
        }
    }

    protected DBHelper getHelper() {
        if (dbHelper == null) {
            dbHelper = (DBHelper) OpenHelperManager.getHelper(this, DBHelper.class);
        }
        return dbHelper;
    }

    @UiThread
    public void showWarning(String message) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }

    public void showWarning(@StringRes int resId) {
        showWarning(getString(resId));
    }

}

File: rock63/src/main/java/com/uksusoff/rock63/ui/BaseActivity.kt
package com.uksusoff.rock63.ui

import android.content.Intent
import android.os.Bundle
import androidx.annotation.StringRes
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.j256.ormlite.android.apptools.OpenHelperManager
import com.uksusoff.rock63.data.DBHelper
import com.uksusoff.rock63.data.InternalPrefs_
import org.androidannotations.annotations.EActivity
import org.androidannotations.annotations.UiThread
import org.androidannotations.annotations.sharedpreferences.Pref

@EActivity
abstract class BaseActivity : AppCompatActivity() {

    private var dbHelper: DBHelper? = null

    @Pref
    protected lateinit var internalPrefs: InternalPrefs_

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        sendBroadcast(Intent(ACTION_CHECK_ALARM))
    }

    override fun onDestroy() {
        super.onDestroy()
        if (dbHelper != null) {
            OpenHelperManager.releaseHelper()
            dbHelper = null
        }
    }

    protected val helper: DBHelper?
        protected get() {
            if (dbHelper == null) {
                dbHelper = OpenHelperManager.getHelper(this, DBHelper::class.java) as DBHelper
            }
            return dbHelper
        }

    @UiThread
    open fun showWarning(message: String?) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
    }

    fun showWarning(@StringRes resId: Int) {
        showWarning(getString(resId))
    }

    companion object {
        const val ACTION_CHECK_ALARM = "com.uksusoff.rock63.ui.ACTION_CHECK_ALARM"
    }
}